core/domain/suggestion_services.py

```diff
@@ -0,0 +1,382 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Funtions to create, accept, reject, update and perform other operations on
+suggestions.
+"""
+
+from core.domain import feedback_services
+from core.domain import suggestion_registry
+from core.domain import user_services
+from core.platform import models
+import feconf
+import utils
+
+(feedback_models, suggestion_models) = models.Registry.import_models([
+    models.NAMES.feedback, models.NAMES.suggestion])
+
+DEFAULT_SUGGESTION_THREAD_SUBJECT = 'Suggestion from a user'
+DEFAULT_SUGGESTION_THREAD_INITIAL_MESSAGE = ''
+
+
+def create_suggestion(
+        suggestion_type, target_type, target_id, target_version_at_submission,
+        author_id, change_cmd, score_category, description,
+        assigned_reviewer_id, final_reviewer_id):
+    """Creates a new SuggestionModel and the corresponding FeedbackThread.
+
+    Args:
+        suggestion_type: str. The type of the suggestion.
+        target_type: str. The target entity being edited.
+
+        (The above 2 parameters should be one of the constants defined in
+        storage/suggestion/gae_models.py.)
+
+        target_id: str. The ID of the target entity being suggested to.
+        target_version_at_submission: int. The version number of the target
+            entity at the time of creation of the suggestion.
+        author_id: str. The ID of the user who submitted the suggestion.
+        change_cmd: dict. The details of the suggestion.
+        score_category: str. The category to score the author on.
+        description: str. The description of the changes provided by the author.
+        assigned_reviewer_id: str(optional). The ID of the user assigned to
```

Comment:
Typeinfo should be "str or None." Ditto below.

(I mean, it's clearly not optional, there's no default arg. And there probably shouldn't be.)

Reply:
Done